thoth/storages/graph/postgres.py

```diff
@@ -3974,13 +3974,16 @@ def sync_analysis_result(self, document: dict) -> None:
         with self._session_scope() as session, session.begin(subtransactions=True):
             if is_external:
                 sw_class = ExternalSoftwareEnvironment
+                python_version = None
             else:
                 sw_class = SoftwareEnvironment
+                python_version = image_name.rsplit("-", maxsplit=1)[1]   # pyXX
+                python_version = python_version[2:3] + "." + python_version[3:]   # take first digit and put . after it
```

Comment:
LGTM.

Just please add some checks here so we fail with corresponding error message. Otherwise, we can sync wrong data.

Reply:
Do you want me to raise an error or just _LOGGER.error(...) and return?

Reply:
raise an error so the transaction is rolled back and no data are persisted